basics/signals/src/index.ts

```diff
@@ -28,62 +26,54 @@ namespace CommandIDs {
 const extension: JupyterFrontEndPlugin<void> = {
   id: 'signals',
   autoStart: true,
-  requires: [ICommandPalette, ILauncher, IMainMenu],
+  optional: [ILauncher],
+  requires: [ICommandPalette, IMainMenu],
   activate: activate
 };
 
 function activate(
   app: JupyterFrontEnd,
   palette: ICommandPalette,
-  launcher: ILauncher,
-  mainMenu: IMainMenu
+  mainMenu: IMainMenu,
+  launcher: ILauncher | null
 ) {
   const manager = app.serviceManager;
   const { commands, shell } = app;
-  let category = 'Tutorial';
+  let category = 'Examples';
 
   // Add launcher
-  launcher.add({
-    command: CommandIDs.create,
-    category: category
-  });
+  if (launcher) {
```

Comment:
Do we cover the case of `optional` dependencies in some other examples?

Otherwise it could be useful to explain why some dependencies are optional and why we need to check for them like this with a guard. But that could be done in a different example.

Reply:
I think, it would make sense to explain that in the launcher or the status bar examples.